RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h

```diff
@@ -0,0 +1,325 @@
+#ifndef CandidatePointSeededTrackingRegionsProducer_h
+#define CandidatePointSeededTrackingRegionsProducer_h
+
+
+#include "RecoTracker/TkTrackingRegions/interface/TrackingRegionProducer.h"
+#include "RecoTracker/TkTrackingRegions/interface/RectangularEtaPhiTrackingRegion.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/Candidate/interface/Candidate.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "DataFormats/Math/interface/normalizedPhi.h"
+
+/** class CandidatePointSeededTrackingRegionsProducer
+ *
+ * eta-phi TrackingRegions producer in directions defined by overlep between Candidate-based and Point-based regions of interest
+ * from collection defined by the "input" and "points" parameters.
+ *
+ * Four operational modes are supported ("mode" parameter):
+ *
+ *   BeamSpotFixed:
+ *     origin is defined by the beam spot
+ *     z-half-length is defined by a fixed zErrorBeamSpot parameter
+ *   BeamSpotSigma:
+ *     origin is defined by the beam spot
+ *     z-half-length is defined by nSigmaZBeamSpot * beamSpot.sigmaZ
+ *   VerticesFixed:
+ *     origins are defined by vertices from VertexCollection (use maximum MaxNVertices of them)
+ *     z-half-length is defined by a fixed zErrorVetex parameter
+ *   VerticesSigma:
+ *     origins are defined by vertices from VertexCollection (use maximum MaxNVertices of them)
+ *     z-half-length is defined by nSigmaZVertex * vetex.zError
+ *
+ *   If, while using one of the "Vertices" modes, there's no vertices in an event, we fall back into
+ *   either BeamSpotSigma or BeamSpotFixed mode, depending on the positiveness of nSigmaZBeamSpot.
+ *
+ *   \author Thomas Strebler
+ */
+class  CandidatePointSeededTrackingRegionsProducer : public TrackingRegionProducer
```

Comment:
@makortel @VinInn please take a look at this one. 
Thank you.

Reply:
@slava77 I replied in the master PR https://github.com/cms-sw/cmssw/pull/20370#issuecomment-328086074.